### PR TITLE
feat(plugin-npm-cli): yarn npm info

### PR DESCRIPTION
**Pull request by paul-soporan**
**What's the problem this PR addresses?**

This PR ports the `yarn info` command over to Berry as `yarn npm info`.

**How did you fix it?**

I ported the `yarn info` command over to Berry, inside `@yarnpkg/plugin-npm-cli` as `yarn npm info`.

Usage, Examples, and Details can be viewed on the deploy preview of the Website.

It works nearly the same as in v1.

Notable exceptions:
- The command is now `yarn npm info` instead of `yarn info`.
- By default, `yarn npm info` will not return the `dist`, `readme`, and `users` fields, since they are often very long. To explicitly request those fields, use the `-v,--verbose` flag or the second argument: `yarn npm info react readme`. This differs from v1, where only the `readme` field was hidden by default and there was no `-v,--verbose` flag.
- The default output format is JSON5, but it can be changed to JSON using the `--json` flag. This differs from v1, where the default output was a JavaScript object. Because JSON5 is very similar to JavaScript objects, there aren't really visual differences.

**Edit**: Changed `yarn info` to `yarn npm info`.

### CODE REVIEW

**Comment by paul-soporan** [comment]
As @arcanis suggested on Discord, I moved `yarn info` from `@yarnpkg/plugin-info` to `@yarnpkg/plugin-npm-cli` as `yarn npm info`.

**Comment by arcanis** [comment]
Nice! I've made a few tweaks to bring it more in line with how other commands work, and taking advantage of the major version increase to make it slightly more powerful. Does that look good for you? In particular:

- We can now request the information from multiple packages in one pass
- We can now request multiple fields in one pass
- By default the display is coloured
- The long fields (`readme` etc) will now be printed only if `--json` or `--fields` are set
- No more verbose flag (it's kinda duplicate with `--json`)

File: packages/plugin-npm-cli/sources/commands/npm/info.ts [hunk 1]
```diff
@@ -17,130 +17,135 @@ declare module '@npm/types' {
...
-      if (!workspace)
-        throw new WorkspaceRequiredError(project.cwd, this.context.cwd);
+    const fields = typeof this.fields !== `undefined`
+      ? new Set(this.fields.split(/\s*,\s*/))
+      : null;
 
-      if (!workspace.manifest.name)
-        throw new UsageError(`Missing \`name\` field in ${path.join(workspace.cwd, `package.json`)}`);
+    if (fields !== null)
+      fields.add(`name`);
```
Comment (paul-soporan) [inline packages/plugin-npm-cli/sources/commands/npm/info.ts]: Wouldn't it be better if the first field that is shown is always `name`?

File: packages/plugin-npm-cli/sources/commands/npm/info.ts [hunk 2]
```diff
@@ -17,130 +17,135 @@ declare module '@npm/types' {
...
     details: `
-      This command will fetch information about a package and return it in a tree format.
-      The package does not have to have been installed locally.
+      This command will fetch information about a package from the npm registry, and prints it in a tree format.
 
-      The default reporting style for this command is a single-quoted serialization (JSON5). To emit lines of valid JSON, use the \`--json\` flag.
+      The package does not have to be installed locally, but needs to have been published (in particular, local changes will be ignored even for workspaces).
 
-      Append \`@[version]\` to the package argument to provide information specific to that version. If the version is invalid, the command falls back to latest.
+      Append \`@<version>\` to the package argument to provide information specific to that version. If the version is invalid, the command will print a warning and fall back to latest.
```
Comment (paul-soporan) [inline packages/plugin-npm-cli/sources/commands/npm/info.ts]: The command doesn't currently print a warning if the version is invalid.

**Comment by paul-soporan** [comment]
Looks good! There are 2 things that I pointed out in the comments. This is a huge improvement from my original PR. Thank you!😃

File: packages/plugin-npm-cli/sources/commands/npm/info.ts (hunk 2 above)
Comment (paul-soporan) [reply packages/plugin-npm-cli/sources/commands/npm/info.ts]: This is something I can implement, but only in a few hours.

File: packages/plugin-npm-cli/sources/commands/npm/info.ts (hunk 1 above)
Comment (arcanis) [reply packages/plugin-npm-cli/sources/commands/npm/info.ts]: Sounds good to me 👍

File: packages/plugin-npm-cli/sources/commands/npm/info.ts (hunk 2 above)
Comment (arcanis) [reply packages/plugin-npm-cli/sources/commands/npm/info.ts]: Sounds good! No rush 👍

**Comment by paul-soporan** [comment]
In my latest commit I experimented with specifying ranges instead of versions since there is enough information to find the highest version that satisfies a range. What do you think, @arcanis?

**Comment by arcanis** [comment]
All clear, thanks Paul!

**Pull request merged**